src/parser.rs

```diff
@@ -2044,17 +2046,223 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_create_role(&mut self) -> Result<Statement, ParserError> {
+        let if_not_exists = self.parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
+        let names = self.parse_comma_separated(Parser::parse_object_name)?;
+
+        let _ = self.parse_keyword(Keyword::WITH);
```

Comment:
As written this ignores errors. I think it should be

```suggestion
        self.parse_keyword(Keyword::WITH)?;
```

Reply:
Sorry -- I misunderstood this code. It actually skips the optional `WITH` clause. Added a PR with a test https://github.com/sqlparser-rs/sqlparser-rs/pull/627